
# Long-Course/Bootcamp

**Target Audience:** Biology/Life Sciences Students

**Goal: to provide a wider, deeper computational understanding of high dimensional data analysis (specifically, NGS data).** Getting students to develop computational thinking skills in the context of answering biological questions. As we start to asssembe the curriculum we can refer to a variety of available course materials.

## Useful Resources for Training in Bioinformatics

**Stephen Turner's RNA-Seq data analysis course** 
Duration: one-day workshop using Amazon instance
SWC-like materials for an Intro to Unix/Shell and possibly R. Walk-through analysis from raw reads to differential expression. 
http://bioconnector.github.io/workshops/lessons/rnaseq-1day/
http://gettinggeneticsdone.blogspot.com/2014/11/rna-seq-data-analysis-course-materials.html

**Titus Brown's NGS Course** 
Duration: two-week workshop using Amazon instance
Emphasis on variety of NGS methods (mapping, RNA-Seq, variant calling) including small intro with BLAST. Topics from SWC are short and interspersed throughout. Good coverage on NGS topics.
http://angus.readthedocs.org/en/2014/

**edX Course on NGS topics using R** 
MOOCS, online 8-course series
http://simplystatistics.org/2014/11/25/harvardx-biomedical-data-science-open-online-training-curriculum-launches-on-january-19/
1. Statistics and R for the Life Sciences
2. Introduction to Linear Models and Matrix Algebra
3. Advanced Statistics for the Life Sciences
4. Introduction to Bioconductor
5. Case study: RNA-seq data analysis
6. Case study: Variant Discovery and Genotyping
7. Case study: ChIP-seq data analysis
8. Case study: DNA methylation data analysis

**Canadian Bioinformatics Workshops**
A collection of two-day workshops covering a wide-aray of topics including NGS, R, Pathway and Network analysis.
http://bioinformatics.ca/workshops/2014
 
**Statistical/data analysis Course material using R**
A _stats course at UBC_ run by Jenny Bryan @Jenny Bryan. Topics covered include Git, data manipulation (tidy data) and graph/figures. Also useful notes on creating R packages.
http://stat545-ubc.github.io/topics.html

_EdX course_ run by Rafael Irizzary and Mike Love. More focus on microarray and some lectures are incomplete.
http://genomicsclass.github.io/book/

_swirl:_ interactively learn R. Using R Studio you will be prompted in the console and explanations provided as you proceed. A variety of course options at the beginner, intermediate and advanced levels. 
http://swirlstats.com/

The advanced statistical inference course follows closely with a course at JHU:
https://github.com/DataScienceSpecialization/courses

Or create your own swirl course with interatcive content of your choice:
http://swirlstats.com/instructors.html

**PacBio turtorial series**
Workshops and slides are PacBio specific. Details on teh technology/chemistry and expected results.
https://github.com/PacificBiosciences/Bioinformatics-Training/wiki

**Understanding the algorithms that underlie bioinformatics tools**
http://www.homolog.us/Tutorials/

**Software Carpentry**
Teaching lab for basic scientific computing skills. Topics likely to incorporate include Unix Shell, Git/Version Control, R and Python 
http://software-carpentry.org/lessons.html

**BITS tutorial series for using UCSC Genome Browser**
Countway library provides helpful video tutorials on using the UCSC Genome Browser, BLAST, and otehr NCBI Resources
https://www.countway.harvard.edu/menuNavigation/libraryServices/classes/videoTutorials.html

**Genome Education Partnership** 
http://gep.wustl.edu/
* In collaboration with the Genome Center of Washington University in St. Louis
* Provide helpful video tutorials on different aspects of NGS; difference between technologies (old versus new), library prep, amplification
* Scripts with slides available: http://gep.wustl.edu/curriculum/course_materials_WU/introduction_to_genomics/nextgen_materials

**Github and Version control**
* http://git-scm.com/book/en/v2

**Introduction to Computational Clusters**
For HMS most students will be using the Orchestra cluster: https://wiki.med.harvard.edu/pub/Orchestra/UserTraining/Intro_Orchestra.pdf
For accesss and use of the Odyssey cluster at FAS: https://software.rc.fas.harvard.edu/training/workshop_intro_cluster/latest/#(1)

	
