---
Date: 2013-05-26 14:35
Title: ChIP-Seq: Enrichment analysis
Author: Shannan Ho Sui & Oliver Hofmann
Contact: ohofmann@hsph.harvard.edu
Type: post
Tags: galaxy, chip-seq, meme, great
---

---

Motif and functional enrichment analyses are unlikely to give reliable results given that you are only looking at 32.8 Mb of reads mapping to chr12. To obtain more signal, we will use the complete set of Nanog peak calls generated by ENCODE and look at binding sites for the _whole_ of chr12. This will also keep the sequence data size manageable for this tutorial.

## ChIP-Seq: Motif enrichment analysis

Start by importing `Encode-hesc-Nanog.narrowPeak` from the Galaxy “Data Library” in the “ENCODE peak calls” folder. It is in [_narrowPeak_ format](https://genome.ucsc.edu/FAQ/FAQformat.html#format12) and contains Nanog peak calls for the whole genome. The format is an extension of the BED format and was developed to describe peaks of signal enrichment for pooled, normalized data. 

Convert this file to `BED` format by editing the attributes and selecting the `Convert Format` tab. Select `Convert Genomic Intervals to BED` and click `Convert`. 

To perform motif enrichment, we need to retrieve the genomic DNA sequences for the set of peaks. To do this, you will also need the `hg19-chr12-reference_sequence.fa` we used previously for the alignments. Import this file into your current history from either a previous history or the Data Library. Recall that it was located in Sequence and reference data in the Shared Data library.

1. Click on `Fetch Sequences` followed by `Extract Genomic DNA`. 
2. Select the `Encode-Nanog.narrowPeak` file.
3. Change the `Source for Genomic Data` to History, and then select `hg19-chr12-reference_sequence.fasta`.
4. Click `Execute` to generate a FASTA file containing the genomic DNA given by the coordinates in the file. 

Note that only the chr12 regions are generated because we used the chr12 reference sequence. Download this file to your computer for use in the motif over-representation analysis.

To identify over-represented motifs, we will use [DREME](http://meme.nbcr.net/meme/cgi-bin/dreme.cgi) from the MEME suite of sequence analysis tools. DREME is a motif discovery algorithm designed to find short, core DNA-binding motifs of eukaryotic transcription factors and is optimized to handle large ChIP-seq data sets. 

1. Visit the [DREME website](http://meme.nbcr.net/meme/cgi-bin/dreme.cgi) and select the downloaded Nanog peaks as input to DREME
2. Enter your email address so that DREME can email you once the analysis is complete
3. Enter a job description so you will recognize which job has been emailed to you and then start the search.

You will be shown a status page describing the inputs and the selected parameters, as well as a link to the results at the top of the screen. Clicking the link displays an output page that is continually updated until the analysis has completed. DREME will also email you once the analysis is complete. This may take some time depending on the server load and the size of the file. While you wait, take a look at the expected result (obtained by clicking on the HTML output link once the analysis has completed).

[![DREME](https://dl.dropboxusercontent.com/u/407047/Blog/Screenshots/Chip_Dreme.png)](https://dl.dropboxusercontent.com/u/407047/Blog/Screenshots/Chip_Dreme.png)

DREME’s HTML output provides a list of **Discovered Motifs** displayed as sequence logos (in the forward and reverse complement (RC) orientations), along with an E-value for the significance of the result. Clicking on `More` displays the number of times the motif was identified. 

To determine if the identified motifs resemble the binding motifs of known transcription factors, we can submit them to [Tomtom](http://meme.nbcr.net/meme/cgi-bin/tomtom.cgi), which searches a database of known motifs to find potential matches and provides a statistical measure of motif-motif similarity. We can run the analysis individually for each motif prediction, or analyze them all at once.

1. Scroll to the bottom of the motif list and click `Submit All`. A dialog box will appear asking you to `Select what you want to do` or `Select a program`. Select `Tomtom` and click `Send`.
2. This takes you to the input page. Tomtom allows you to select the database you wish to search against. We will use the default but keep in mind that there are other options when performing your own analyses. 
3. Enter your email address and job description and start the search.

You will be shown a status page describing the inputs and the selected parameters, as well as a link to the results at the top of the screen. Clicking the link displays an output page that is continually updated until the analysis has completed. Like DREME, Tomtom will also email you the results.

The HTML output for Tomtom shows a list of possible motif matches for each DREME motif prediction generated from your Nanog regions. Clicking on the match name shows you an alignment of the predicted motif versus the match.

> What are the possible identities for the motifs you discovered from the Nanog ChIP regions on chr12? Do these make sense given what you know about the Nanog transcription factor?  Compare your results to results for the full ENCODE data set available from [FactorBook](http://www.factorbook.org/mediawiki/index.php/NANOG) and take a look at a [Nanog binding matrix](http://rsat.ulb.ac.be/data/published_data/Medina-Rivera_2010//Mus_musculus/chip-seq_Chen_2008/results/matrix_quality/Nanog/Nanog_M01247.tf/Nanog_M01247.tf_synthesis.html) (from [Ferraris _et. al._, 2013](http://genome.cshlp.org/content/early/2011/04/27/gr.115824.110.full.pdf+html)).


## Functional enrichment

A large number of methods and external tools are available to make sense of a gene list, most of which revolve around detecting functional enrichment in Gene Ontology, curated pathways or networks. A recent post on [Getting Genetics Done](http://gettinggeneticsdone.blogspot.com/2012/03/pathway-analysis-for-high-throughput.html) should get you started, and we will expand this section for future workshops. For now we will explore one approach that cover the basics.

In order to be able to identify functional enrichment we need to look at genome wide signals; instead of just looking at the ENCODE Nanog peak calls for chromosome 12 we will use the full set of calls. We will use [GREAT](http://bejerano.stanford.edu/great/public/html/index.php) to perform a functional enrichment. GREAT takes a list of regions, associates them with nearby genes, and then analyzes the gene annotations to assign biological meaning to the data:

1. Go back to Galaxy and download the BED file you created from the `Encode-Nanog.narrowPeak` file for the motif analysis. Make sure it is the BED formatted file as this is a requirement for the next step.
2. First select `Human: GRCh37 (UCSC hg19, Feb/2009)` for the `Species Assembly`.
3. Choose the ENCODE-Nanog.bed file and use the `Whole genome` for `Background regions`.
4. Click `Submit`.

GREAT provides the output in HTML format organized by section. 

[![GREAT](https://dl.dropboxusercontent.com/u/407047/Blog/Screenshots/Chip_Great.png)](https://dl.dropboxusercontent.com/u/407047/Blog/Screenshots/Chip_Great.png)

The first section contains the **Job Description** detailing the inputs and parameters. The **Region-Gene Association Graphs** indicate the number of associated genes per region, and the distances of the regions to transcription start sites (TSS), providing insights into the genomic organization of Nanog ChIP regions.

> Explore the rest of the results. The output contains enrichment of GO terms, pathway / disease enrichments, miRNA and TF-targets. It is also worthwhile to click on the enriched annotation names to get a more detailed view of genes annotated with these terms.


## We need your feedback

In order to fund continued development of these courses we rely on your criticism and feedback. _Please_ do take 2-3 minutes to fill our [**the survey**](http://www.surveymonkey.com/s/5VM9HZD), but also let us know in person if you have questions or suggestions.